core/src/main/scala/caliban/parsing/VariablesUpdater.scala

```diff
@@ -16,38 +21,75 @@ object VariablesUpdater {
   ): GraphQLRequest = {
     val variableDefinitions = doc.operationDefinitions.flatMap(_.variableDefinitions)
     val updated             = req.variables.getOrElse(Map.empty).map { case (key, value) =>
-      val v = variableDefinitions.find(_.name == key).map(resolveEnumValues(value, _, rootType)).getOrElse(value)
+      val v =
+        variableDefinitions
+          .find(_.name == key)
+          .map { (definition: VariableDefinition) =>
```

Comment:
Why do you need to specify the type?

Reply:
Ah, I didn't anymore. Was likely an earlier compilation error. Well spotted! 👀